classic/form/field/GeocoderComboBox.js

```diff
@@ -214,10 +221,67 @@ Ext.define('GeoExt.form.field.GeocoderComboBox', {
         me.callParent(arguments);
 
         me.on({
-            select: this.onSelect,
+            unRestrictMapExtent: me.unRestrictExtent,
+            restrictToMapExtent: me.restrictExtent,
+            select: me.onSelect,
             focus: me.onFocus,
             scope: me
         });
+
+        if (me.restrictToMapExtent) {
+            me.restrictExtent();
+        }
+    },
+
+    /**
+     * Handle restriction to viewbox: register moveend event
+     * and update params of AJAX proxy
+     */
+    restrictExtent: function() {
+        var me = this;
+        me.map.on('moveend', me.updateExtraParams, me);
+        me.updateExtraParams();
+    },
+
+    /**
+     * Update viewbox parameter based on the current map extent
+     */
+    updateExtraParams: function() {
+        var me = this;
+        var mapSize = me.map.getSize();
+        var mv = me.map.getView();
+        var extent = mv.calculateExtent(mapSize);
```

Comment:
I suggest to move the code below to an extra function,which retrieves the extent as parameter. Since the code of this class should work with various Geocoder providers it should be easily overwritable (as done with address bbox (`convertToExtent`) or address point (`convertToCoordinate`) parsing). The new function can have the Nominatim implementation you provided as default.

Something llike:

```
        addMapExtentParams: function() {/** your code for nominatim */}
```

Or we decide to let the user overwrite `updateExtraParams` at all. Up to you.

Reply:
:+1: 

I've moved the update of the AJAX proxy `url` parameters to a new function called `addMapExtentParams', as you suggested. Here, I've also added the projection as an optional parameter. So the user can pass an extent that has not necessarily the projection of the map.

Reply:
Nice, thanks!